client/extensions/woocommerce/woocommerce-services/views/packages/add-package.js

```diff
@@ -0,0 +1,245 @@
+			action: 'delete',
+			label: <span><Gridicon icon="trash" /> { translate( 'Delete this package' ) }</span>,
+			onClick: onRemove,
+			additionalClassNames: 'packages__delete is-scary is-borderless'
+		} );
+	}
+
+	return buttons;
+};
+
+const OuterDimensionsToggle = ( { siteId, toggleOuterDimensions, translate } ) => {
+	const onClick = ( evt ) => {
+		evt.preventDefault();
+		toggleOuterDimensions( siteId );
+	};
+
+	return (
+		<a href="#" className="packages__setting-explanation" onClick={ onClick }>
+			{ translate( 'View exterior dimensions' ) }
+		</a>
+	);
+};
+
+const AddPackageDialog = ( props ) => {
+	const {
+		siteId,
+		form,
+		dismissModal,
+		toggleOuterDimensions,
+		setModalErrors,
+		savePackage,
+		updatePackagesField,
+		removePackage,
+		translate,
+	} = props;
+
+	const {
+		showModal,
+		mode,
+		modalErrors,
+		dimensionUnit,
+		weightUnit,
+		packages,
+		packageSchema,
+		predefinedSchema,
+		packageData,
+		showOuterDimensions,
+	} = form;
+
+	const customPackages = packages.custom;
+
+	const {
+		index,
+		name,
+		inner_dimensions,
+		outer_dimensions,
+		box_weight,
+		max_weight,
+		is_user_defined,
+	} = packageData;
+
+	const isOuterDimensionsVisible = showOuterDimensions || outer_dimensions;
+	const exampleDimensions = [ 100.25, 25, 5.75 ].map( ( val ) => val.toLocaleString() ).join( ' x ' );
+
+	const onSave = () => {
+		const editName = 'number' === typeof packageData.index ? customPackages[ packageData.index ].name : null;
+
+		//get reserved box names:
+		const boxNames = _.concat(
+			_.difference( customPackages.map( ( boxPackage ) => boxPackage.name ), [ editName ] ), //existing custom boxes
+			_.flatten( _.map( predefinedSchema, predef => _.map( predef, group => group.definitions ) ) ), //predefined boxes
+			[ 'individual' ] //reserved for items shipping in original packaging
+		);
+
+		const filteredPackageData = Object.assign( {}, packageData, {
+			name: inputFilters.string( packageData.name ),
+			inner_dimensions: inputFilters.dimensions( packageData.inner_dimensions ),
+			outer_dimensions: inputFilters.dimensions( packageData.outer_dimensions ),
+			box_weight: inputFilters.number( packageData.box_weight ),
+			max_weight: inputFilters.number( packageData.max_weight ),
+		} );
+
+		const errors = checkInputs( filteredPackageData, boxNames, packageSchema );
+		if ( errors.any ) {
+			updatePackagesField( siteId, filteredPackageData );
+			setModalErrors( siteId, errors );
+			return;
+		}
+
+		savePackage( siteId, filteredPackageData );
+	};
+
+	const updateTextField = ( event ) => {
+		const key = event.target.name;
+		const value = event.target.value;
+		setModalErrors( siteId, _.omit( modalErrors, key ) );
+		updatePackagesField( siteId, { [ key ]: value } );
+	};
+
+	const fieldInfo = ( field, nonEmptyText ) => {
+		const altText = nonEmptyText || translate( 'Invalid value' );
+		const text = '' === _.trim( packageData[ field ] ) ? translate( 'This field is required' ) : altText;
```

Comment:
Hi! I've found a possible matching string that has already been translated **39** times:
`translate( 'This field is required.' )` *ES Score: 15*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/16409-packages-settings)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).